Week4/app/src/main/java/com/em4n0101/mymovies/MovieDetailFragment.kt

```diff
@@ -0,0 +1,60 @@
+package com.em4n0101.mymovies
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.ViewModelProvider
+import com.em4n0101.mymovies.data.Movie
+import com.em4n0101.mymovies.utils.Utilities
+import kotlinx.android.synthetic.main.fragment_movie_detail.*
+
+class MovieDetailFragment : Fragment() {
+
+    private lateinit var moviesManager: MoviesViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_movie_detail, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        arguments?.let {
+            // get the title of the movie to display
+            val args = MovieDetailFragmentArgs.fromBundle(it)
+
+            // get view model
+            moviesManager = ViewModelProvider(this).get(MoviesViewModel::class.java)
+            val movie = moviesManager.getListOfMovies().filter { movie -> movie.title == args.movieTitleString }[0]
```

Comment:
It would be better if you did this filtering inside of the ViewModel. Try to have as less logic as possible in your Fragments.